Practical7/src/store/index.js

```diff
@@ -0,0 +1,173 @@
+      commit("setApiError", state, payload),
+    setUserAuth: ({ commit }, state, payload) =>
+      commit("setUserAuth", state, payload),
+    setForm: ({ commit }, state, payload) => commit("setForm", state, payload),
+
+    async fetchCarDetails({ commit }) {
+      commit("setLoading", true);
+      await axios
+        .get("https://testapi.io/api/dartya/resource/cardata")
+        .then((res) => {
+          commit("setCars", res.data.data);
+          commit("setLoading", false);
+        })
+        .catch((err) => {
+          commit("setApiError", err);
+          commit("setLoading", false);
+        });
+    },
+
+    async postCarDetails({ commit, dispatch }, payload) {
+      commit("setLoading", true);
+      await axios
+        .post("https://testapi.io/api/dartya/resource/cardata", payload)
+        .then((res) => {
+          if (res.status === 201) {
+            dispatch("fetchCarDetails");
+          }
+          commit("setLoading", false);
+          commit("setForm", (this.state.formDetails.showForm = false));
+        })
+        .catch((err) => {
+          commit("setLoading", false);
+          commit("setForm", (this.state.formDetails.showForm = false));
+          alert("can't do the operation at this moment" + err);
+        });
+    },
+
+    async editCarDetails({ commit, dispatch }, payload) {
+      commit("setLoading", true);
+      await axios
+        .put(
+          `https://testapi.io/api/dartya/resource/cardata/${payload.id}`,
+          payload
+        )
+        .then((res) => {
+          if (res.status === 200) {
+            dispatch("fetchCarDetails");
+          }
+          commit("setLoading", false);
+          commit("setForm", (this.state.formDetails.showForm = false));
+        })
+        .catch((err) => {
+          commit("setLoading", false);
+          commit("setForm", (this.state.formDetails.showForm = false));
+          alert("can't do the edit at this moment" + err);
+        });
+    },
+
+    async deleteCarDetails({ commit, dispatch }, payload) {
+      commit("setLoading", true);
+      await axios
+        .delete(`https://testapi.io/api/dartya/resource/cardata/${payload}`)
+        .then((res) => {
+          if (res.status === 204) {
+            dispatch("fetchCarDetails");
+          }
+          commit("setLoading", false);
+        })
+        .catch((err) => {
+          alert("can't delete at this moment" + err);
+        });
+    },
+
+    async loginUser({ commit }, payload) {
+      commit("setLoading", true);
+      await axios
+        .get("https://testapi.io/api/dartya/resource/users")
+        .then((res) => {
+          let allusers = res.data.data;
+          let user = allusers.find((user) => {
+            return (
+              user.email == payload.email && user.password == payload.password
+            );
+          });
+          commit("setLoading", false);
+          if (user == undefined) {
+            commit("setApiError", true);
```

Comment:
why used commit 2 times

Reply:
Both setLoading and setApiError are different. and if apierror will be true then on loading page error div will show thats why.